lxd/device/pci.go

```diff
@@ -0,0 +1,122 @@
+package device
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	deviceConfig "github.com/lxc/lxd/lxd/device/config"
+	pcidev "github.com/lxc/lxd/lxd/device/pci"
+	"github.com/lxc/lxd/lxd/instance"
+	"github.com/lxc/lxd/lxd/instance/instancetype"
+	"github.com/lxc/lxd/shared/validate"
+)
+
+type pci struct {
+	deviceCommon
+}
+
+// validateConfig checks the supplied config for correctness.
+func (d *pci) validateConfig(instConf instance.ConfigReader) error {
+	if !instanceSupported(instConf.Type(), instancetype.VM) {
+		return ErrUnsupportedDevType
+	}
+
+	rules := map[string]func(string) error{
+		"address": validate.IsPCIAddress,
+	}
+
+	err := d.config.Validate(rules)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to validate config")
+	}
+
+	// PCI devices can be specified as "0000:XX:XX.X" or "XX:XX.X".
```

Comment:
@stgraber this pci logic has been repeated a few times already in the code base, I wonder if it would be useful to be move this to the `pci` package as `pcidev.NormaliseAddress()` or `pcidev.CanonicalAddress()`?